.github/workflows/check-pr-description.yaml

```diff
@@ -0,0 +1,42 @@
+name: "Check PR description"
```

Comment:
(since I was added as reviewer) I haven't looked any deep in the code so I'd need another pass (or a few) in order to provide detailed comments. However, at this point I have a few general questions.

1. @tommartensen [was considering](https://github.com/stackrox/stackrox/pull/11318#discussion_r1621284399) to add `Shopify/task-list-checker` to validate checkboxes. I also saw [ROX-24344](https://issues.redhat.com/browse/ROX-24344). I'm not opposed to hand-rolled approach but it is not good if we do one thing in different ways in multiple places. It would be much better to use one approach (unless there are fundamental differences). Do you know the sate of Tom's efforts?
2. I have an impression that `Check PR description` should show up in GitHub status for this PR but it's not on the list and I can't spot it here <https://github.com/stackrox/stackrox/actions?query=branch%3Amichael%2Fcheck-pr-health-workflow+>. Am I looking for a wrong thing or in a wrong place?
3. Maybe I'm not used to GHA enough but I find it inconvenient to navigate them in the tiny scrollable status window (see screenshot) and I find it inconvenient that clicking Details somewhere takes me to a page for that workflow with no ability to transition to a different workflow for the same revision in a PR. (What I mean is: click for example [here](https://github.com/stackrox/stackrox/actions/runs/9713757062/job/26811233510?pr=11716) and try to get to Style workflow run.) In my opinion, it would make things a little more convenient if you could group smaller workflows that deal with checking/altering PRs in one. This check, `Check PR title`, `Label pull requests based on changed files` seem to easily belong the same group. WDYT?

![image](https://github.com/stackrox/stackrox/assets/537715/7d2ef29d-0eb1-4a09-b7f3-40f768f45d0c)

Reply:
1. I looked at various 3rd party options, and found that they are less flexible than a simple script like in this PR, some require special annotations in the PR template, which I don't like, and some don't check for random prohibited patterns like 'change me!'. In the end, configuring a 3rd party action may be more complex, than adding a simple condition to the custom script.
2. You're looking at the wrong place (repository). As this is a new workflow, it doesn't exist on master, and hence not yet executed. I tested it on stackrox/test-gh-actions repo: https://github.com/stackrox/test-gh-actions/actions/workflows/check-pr-description.yaml
3. The checks are also available at the [Checks](https://github.com/stackrox/test-gh-actions/pull/178/checks) PR tab, where one can find them grouped by workflow job.

Reply:
1. My question was how far along was @tommartensen in the implementation of ROX-24344. The task status is `New`, I don't find mentions of `task-list-checker` in the `master` branch. Looks like we should assume it's safe to go forward with this PR and that it does not duplicate efforts and does not introduce another-way-of-doing-the-same-thing.
2. Is there no way to run a new workflow in the same PR that introduces it in GHA? I met some CI systems that have such a limitation and thought it's a silly one (although, of course, technically conditioned by something), I'm surprised that GHA is of this kind. Folks [here](https://stackoverflow.com/a/76439748/484050) and [here](https://stackoverflow.com/questions/63362126/github-actions-how-can-i-run-a-workflow-created-on-a-non-master-branch-from-t) suggest it should be possible.
3. TIL about the Checks tab, never bothered to look there. Indeed, it's more handy. My question about grouping related jobs in one workflow file is still open.

Reply:
1. Not sure @tommartensen has committed anything yet for ROX-24344. He suggested me to check out the alternatives.
2. Today I remembered, that it is indeed possible to test the new workflow from a branch, if using `pull_request` trigger instead of `pull_request_target`. See a run of the workflow under review on this same PR: https://github.com/stackrox/stackrox/actions/runs/9836140777?pr=11716
3. Putting various logically related workflows to the same file may indeed provide better visual grouping and navigation. We could indeed move linting, code and PR styles to the same workflow file. Lemme do it in a separate PR.

Reply:
1. For me it was important to know that you talked to each other.
3. Deal.